python_modules/libraries/dagster-teradata/dagster_teradata/resources.py

```diff
@@ -0,0 +1,464 @@
+            if teradata_authorization_name:
+                credentials_part = f"AUTHORIZATION={teradata_authorization_name}"
+            else:
+                # Obtaining Azure client ID and secret from the azure_blob resource
+                credentials_part = (
+                    f"ACCESS_ID= '{azure_client_id}' ACCESS_KEY= '{azure_client_secret}'"
+                )
+
+        sql = dedent(f"""
+                    CREATE MULTISET TABLE {teradata_table} AS
+                    (
+                        SELECT * FROM (
+                            LOCATION = '{blob_source_key}'
+                            {credentials_part}
+                        ) AS d
+                    ) WITH DATA
+                    """).rstrip()
+
+        self.execute_queries(sql)
+
+    # Handler to handle single result set of a SQL query
+    def _single_result_row_handler(self, cursor):
+        records = cursor.fetchone()
+        if isinstance(records, list):
+            return records[0]
+        if records is None:
+            return records
+        raise TypeError(f"Unexpected results: {cursor.fetchone()!r}")
+
+    def create_teradata_compute_cluster(
+        self,
+        compute_profile_name: str,
+        compute_group_name: str,
+        query_strategy: str = "STANDARD",
+        compute_map: str = None,
+        compute_attribute: str = None,
+        timeout: int = constants.CC_OPR_TIME_OUT,
+    ):
+        return self.compute_cluster_manager.create_teradata_compute_cluster(
+            compute_profile_name,
+            compute_group_name,
+            query_strategy,
+            compute_map,
+            compute_attribute,
+            timeout,
+        )
+
+    def drop_teradata_compute_cluster(
+        self,
+        compute_profile_name: str,
+        compute_group_name: str,
+        delete_compute_group: bool = False,
+    ):
+        return self.compute_cluster_manager.drop_teradata_compute_cluster(
+            compute_profile_name,
+            compute_group_name,
+            delete_compute_group,
+        )
+
+    def resume_teradata_compute_cluster(
+        self,
+        compute_profile_name: str,
+        compute_group_name: str,
+    ):
+        return self.compute_cluster_manager.resume_teradata_compute_cluster(
+            compute_profile_name,
+            compute_group_name,
+        )
+
+    def suspend_teradata_compute_cluster(
+        self,
+        compute_profile_name: str,
+        compute_group_name: str,
+    ):
+        return self.compute_cluster_manager.suspend_teradata_compute_cluster(
+            compute_profile_name,
+            compute_group_name,
+        )
+
+
+@dagster_maintained_resource
+@resource(
+    config_schema=TeradataResource.to_config_schema(),
+    description="This resource is for connecting to the Teradata Vantage",
+)
+def teradata_resource(context) -> TeradataConnection:
```

Comment:
Does this method used internal by dagster core framework?

Reply:
Yes, this is mandatory